Source/objects.cpp

```diff
@@ -725,7 +725,22 @@ void AddChestTraps()
 	}
 }
 
-void LoadMapObjects(const char *path, int startx, int starty, int x1, int y1, int w, int h, int leveridx)
+/**
+ * @brief Used by the LoadMapObjects routines to translate from "grid" coordinates to "tile" coordinates
+ *
+ * This appears to be mainly to handle differences in map scale for certain environments where some are
+ * based on 2x2 objects and some are 1x1.
+ * 
+ * @param start Initial location in grid coordinates
+ * @param offset Distance from the grid origin
+ * @return start + offset + a constant factor of {16, 16}
+*/
+constexpr Point convertToMapPos(Point start, Displacement offset)
+{
+	return start + Displacement { 16, 16 } + offset;
+}

```

Comment:
What's happening here is not a conversion from the MegaTiles to DungeonPiece scale. It is simply adding the 16dp border (for rendering) and displacing the map data to the given `start` location.

The 2x2 mega tile only applies to the dimensions in the .dun header and the size of the first of 4 layers (tile, monster, item, object).

Pseudo format definition for .dun:
```
typedef MegaTiles uint16_t;
typedef DungeonPiece uint16_t;
dun {
    uint16_t mtWidth;
    uint16_t mtHeight;
    MegaTiles tiles[mtWidth][mtHeight];
    DungeonPiece unused[mtWidth*2][mtHeight*2];
    DungeonPiece monsters[mtWidth*2][mtHeight*2];
    DungeonPiece objects[mtWidth*2][mtHeight*2];
    DungeonPiece transparancy[mtWidth*2][mtHeight*2];
}
```

The only time the conversion is relevant is for calculating how much data to skip to locate the `objectLayer` portion of the data.

Instead I would suggest adding the 16dp border to the start location outside the loop

Reply:
Did a bunch of edits to try and make the comment more clear :)

Reply:
Thanks, I wrote this a while back and had a pretty poor understanding of the reason behind the translation. Your comment makes sense. I've removed the function in the latest commit since it didn't serve much purpose and the doxygen comment was wildly inaccurate.